clang/test/SemaCXX/warn-loop-analysis.cpp

```diff
@@ -299,3 +299,18 @@ void test10() {
   for (auto[i, j, k] = arr; i < a; ++i) { }
   for (auto[i, j, k] = arr; i < a; ++arr[0]) { }
 };
+
+extern void foo(int);
```

Comment:
We normally wrap tests from bug reports in `namespace GHXXXX` where XXXX is the bug report number.

Reply:
Can we also get tests that show diagnostics for lambda cases as well.

Reply:
@shafik, thanks for the review. I've updated tests